astropy/coordinates/representation/base.py

```diff
@@ -486,16 +486,10 @@ def _units(self):
 
     @property
     def _unitstr(self):
-        units_set = set(self._units.values())
-        if len(units_set) == 1:
-            unitstr = units_set.pop().to_string()
-        else:
-            unitstr = "({})".format(
-                ", ".join(
-                    self._units[component].to_string() for component in self.components
-                )
-            )
-        return unitstr
+        units = self._units.values()
+        if len(units_set := set(units)) == 1:
+            return units_set.pop().to_string()
+        return f"({', '.join(unit.to_string() for unit in units)})"
```

Comment:
Since `.to_string()` is the same as `__str__`, one can just do `', '.join(units)`... Perhaps
```
units = self._units.values()
if len(set(units)) == 1:
    return str(units[:1])
return f"({', '.join(units)})"
# Or, maybe clearer, to avoid a still somewhat complicated expression in the f-string
return "(" + ", ".join(units) + ")"
```

Reply:
`str.join()` requires its input to already be an iterable of strings, it does not call `str()` automatically. This means `', '.join(units)` doesn't work, but `', '.join(map(str, units))` does and that is still shorter than `', '.join(unit.to_string() for unit in units)`.

I can agree that having the f-string end with `))})"` is not very appealing, but what is being evaluated inside `{}` is simple enough, what is outside `{}` is well covered by tests and `"(" + ", ".join(map(str, units)) + ")"` is not an improvement in readability.

Reply:
I thought I had tested the `.join()` thing, but I now see I was wrong... I think `map(str, units)` is indeed marginally better than the iterator, but now at the level that it doesn't matter much either way.

And, indeed, the other thing is just a matter of taste, so happy to go with the `f-string`.